### PR TITLE
[NUI] Use correct calling convention

### PR DESCRIPTION
**Pull request by walac**
Some callback and functions from DALi expected the calling convention to
be cdecl, this commit makes it right. The amd64 platforms this is rarely
a problem since arguments are passed through CPU registers, but in x86
this leads to stack corruption.

### CODE REVIEW

**Comment by dongsug-song** [comment]
@walac 
Hello, 
Thank you for your work, but could you move this patch into DevelNUI?
after you have moved this, then I will review and merge it.
thank you.

**Comment by walac** [comment]
@dongsug-song ok, done :)

**Comment by dongsug-song** [comment]
@walac 
could you check again?
I can't merge your patch because of the error shown below.
(the "Rebase and merge" button has not been activated)
thank you.

image

**Comment by walac** [comment]
@dongsug-song done!

**Comment by walac** [comment]
@dongsug-song I believe someone must approve the patch to get the button enabled

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Comment by dongsug-song** [comment]
@walac 
Hello, 
I approved but same error shown, so I can't merge your PR. 
could you check again?
thank you.

**Comment by walac** [comment]
@dongsug-song I updated the patch again, but this is weird, if you have write access to the repository you have be able to merge it.

**Comment by dongsug-song** [comment]
@WonyoungChoi 
Hello, 
I can't merge this PR because "Rebase and merge" button is inactive.
could you merge this patch if possible?
thank you.

**Comment by hyunjushin** [comment]
@WonyoungChoi 
 "Rebase and merge" button is inactive state.
Can you check the reason?

**Review by hyunjushin — APPROVED** [review]

**Comment by rabbitfor** [comment]
Hi @walac 
I see there are many other callback functions declared as `StdCall` in NUI code,
for example,
* Animation.AnimationFinishedEventCallbackType
* TapGestureDetector.DetectedCallbackType

Many DALi callback delegate are defined as `StdCall` now.
Is it OK not to fix them?

In your comment,
> Some callback and functions from DALi expected the calling convention to
be cdecl, this commit makes it right.

I'd like to know what `Some callback and functions`  are.

**Comment by walac** [comment]
Hi @rabbitfor,

> I'd like to know what Some callback and functions are.

If you take a look at the patch, you can see the relevant functions.

> Animation.AnimationFinishedEventCallbackType

> TapGestureDetector.DetectedCallbackType

I only patched the cases I caught while developing, it is possible to have other invalid calls elsewhere.

**Review by AdunFang — APPROVED** [review]
It is OK.

**Comment by dongsug-song** [comment]
@walac 
Hello, 
I made new PR same as yours.
please review this : https://github.com/Samsung/TizenFX/pull/2462
thank you.

**Comment by dongsug-song** [comment]
https://github.com/Samsung/TizenFX/pull/2462
it has been merged, so close this.

**Pull request closed without merging**
